warehouse/integrations/snowflake/snowflake.go

```diff
@@ -576,132 +436,266 @@ func (sf *Snowflake) loadTable(ctx context.Context, tableName string, tableSchem
+	stagingTableName := whutils.StagingTableName(provider, identityMappingsTable, tableNameLimit)
+	sqlStatement := fmt.Sprintf(
+		`CREATE TEMPORARY TABLE %[1]s.%[2]q LIKE %[1]s.%[3]q`,
+		schemaIdentifier, stagingTableName, identityMappingsTable,
+	)
 
-	sf.logger.Infof("SF: Creating temporary table for table:%s at %s\n", identityMappingsTable, sqlStatement)
+	sf.logger.Infow("Creating temporary table", lf.TableName, identityMappingsTable, lf.Query, sqlStatement)
 	_, err = dbHandle.ExecContext(ctx, sqlStatement)
 	if err != nil {
-		sf.logger.Errorf("SF: Error creating temporary table for table:%s: %v\n", identityMappingsTable, err)
-		return
+		sf.logger.Errorw("Error creating temporary table", lf.TableName, identityMappingsTable, lf.Error, err.Error())
+		return fmt.Errorf("cannot create temporary table like %s.%q: %w", schemaIdentifier, identityMappingsTable, err)
 	}
 
-	sqlStatement = fmt.Sprintf(`ALTER TABLE %s.%q ADD COLUMN "ID" int AUTOINCREMENT start 1 increment 1`, schemaIdentifier, stagingTableName)
-	sf.logger.Infof("SF: Adding autoincrement column for table:%s at %s\n", stagingTableName, sqlStatement)
+	sqlStatement = fmt.Sprintf(
+		`ALTER TABLE %s.%q ADD COLUMN "ID" int AUTOINCREMENT start 1 increment 1`,
+		schemaIdentifier, stagingTableName,
+	)
+	sf.logger.Infow("Adding autoincrement column", lf.TableName, stagingTableName, lf.Query, sqlStatement)
 	_, err = dbHandle.ExecContext(ctx, sqlStatement)
 	if err != nil && !checkAndIgnoreAlreadyExistError(err) {
-		sf.logger.Errorf("SF: Error adding autoincrement column for table:%s: %v\n", stagingTableName, err)
-		return
+		sf.logger.Errorw("Error adding autoincrement column", lf.TableName, stagingTableName, lf.Error, err.Error())
+		return fmt.Errorf("cannot add autoincrement column to %s.%q: %w", schemaIdentifier, stagingTableName, err)
 	}
 
-	loadLocation := warehouseutils.GetObjectLocation(sf.ObjectStorage, loadFile.Location)
-	sqlStatement = fmt.Sprintf(`COPY INTO %v("MERGE_PROPERTY_TYPE", "MERGE_PROPERTY_VALUE", "RUDDER_ID", "UPDATED_AT") FROM '%v' %s PATTERN = '.*\.csv\.gz'
-		FILE_FORMAT = ( TYPE = csv FIELD_OPTIONALLY_ENCLOSED_BY = '"' ESCAPE_UNENCLOSED_FIELD = NONE ) TRUNCATECOLUMNS = TRUE`, fmt.Sprintf(`%s.%q`, schemaIdentifier, stagingTableName), loadLocation, sf.authString())
+	loadLocation := whutils.GetObjectLocation(sf.ObjectStorage, loadFile.Location)
+	sqlStatement = fmt.Sprintf(
+		`COPY INTO %s.%q("MERGE_PROPERTY_TYPE", "MERGE_PROPERTY_VALUE", "RUDDER_ID", "UPDATED_AT")
+		FROM '%v' %s PATTERN = '.*\.csv\.gz'
+		FILE_FORMAT = ( TYPE = csv FIELD_OPTIONALLY_ENCLOSED_BY = '"' ESCAPE_UNENCLOSED_FIELD = NONE )
+		TRUNCATECOLUMNS = TRUE`,
+		schemaIdentifier, stagingTableName,
+		loadLocation,
+		sf.authString(),
+	)
 
-	sf.logger.Infof("SF: Dedup records for table:%s using staging table: %s\n", identityMappingsTable, sqlStatement)
+	sf.logger.Infow("Copying identity mappings for table", lf.TableName, identityMappingsTable, lf.Query, sqlStatement)
```

Comment:
Maybe we can:
```suggestion
        logger := sf.logger.With(
        lf.TableName, identityMappingsTable, 
        lf.Query, sqlStatement
       )
	logger.Infow("Copying identity mappings for table")
```

reusing it in the section bellow

Reply:
I can't reuse the `Query` one but I can reuse the table. Making the changes :+1: